x-pack/plugins/observability/public/components/custom_threshold/components/validation.tsx

```diff
@@ -56,14 +59,17 @@ export function validateCustomThreshold({
       buildEsQuery(
         undefined,
         [{ query: (searchConfiguration.query as Query).query, language: 'kuery' }],
-        []
+        [],
+        getEsQueryConfig(uiSettings)
       );
     } catch (e) {
+      const errorReason = e.shortMessage || '';
       errors.filterQuery = [
         i18n.translate(
           'xpack.observability.customThreshold.rule.alertFlyout.error.invalidFilterQuery',
           {
-            defaultMessage: 'Filter query is invalid.',
+            values: { errorReason },
+            defaultMessage: `Filter query is invalid. {errorReason}`,
```

Comment:
Can we have a unit test for it?

Reply:
What do you want to test?

Reply:
Maybe I put the comment on a wrong line, I meant having an invalid query and verifying that the error is generated correctly. (We can have the setting mocked as disabled and check  if invalid message was created corretly.)

Reply:
I realized there were missing tests for other stuff. So, I added tests for filter and query validation.